### PR TITLE
doc: 12.2.1 release notes

### PR DESCRIPTION
**Pull request by theanalyst**
Signed-off-by: Abhishek Lekshmanan <alekshmanan@suse.com>

### CODE REVIEW

**Comment by theanalyst** [comment]
jenkins render docs

File: doc/release-notes.rst [hunk 1]
```diff
@@ -2,6 +2,123 @@
  Release Notes
 ===============
 
+v12.2.1 Luminous
+================
+
+This is the first bugfix release of Luminous v12.2.x long term stable
+release series. We recommend all the users of 12.2.x series update.
```
Comment (smithfarm) [inline doc/release-notes.rst]: Can we model this paragraph on the analogous paragraph from the v10.2.1 release notes?

```
This is the first bugfix release for Jewel.  It contains several annoying
packaging and init system fixes and a range of important bugfixes across
RBD, RGW, and CephFS.

We recommend that all v10.2.x users upgrade.

For more detailed information, see :download:`the complete changelog <changelog/v10.2.1.txt>`.
```

File: doc/release-notes.rst [hunk 2]
```diff
@@ -2,6 +2,123 @@
...
+
+Notable Changes
+---------------
+
+- *RGW*
+
+  * dynamic resharding is now enabled by default, RGW will now automatically
+  reshard the bucket index once the index grows beyond `rgw_max_objs_per_shard`
```
Comment (smithfarm) [inline doc/release-notes.rst]: indent

File: doc/release-notes.rst [hunk 3]
```diff
@@ -2,6 +2,123 @@
...
+
+- *CephFS*:
+
+  * Limiting MDS cache via a memory limit is now supported using the new
+    mds_cache_memory_limit config option (1GB by default).  A cache reservation
+    can also be specified using mds_cache_reservation as a percentage of the
+    limit (5% by default). Limits by inode count are still supported using
+    mds_cache_size. Setting mds_cache_size to 0 (the default) disables the
+    inode limit.
+
```
Comment (smithfarm) [inline doc/release-notes.rst]: Stylistically it's a little weird for the first two bullet points to be under headings (RGW, CephFS) but the last three not.

Also, headings are usually dispensed with when there is only one bullet point under them. Can you integrate the headings "RGW", "CephFS" into the text of the bullet points?

File: doc/release-notes.rst [hunk 4]
```diff
@@ -2,6 +2,123 @@
...
+* core: incorrect erasure-code space in command ceph df (`issue#21243 <http://tracker.ceph.com/issues/21243>`_, `pr#17724 <https://github.com/ceph/ceph/pull/17724>`_, liuchang0812)
+* core: interval_set: optimize intersect_of insert operations (`issue#21229 <http://tracker.ceph.com/issues/21229>`_, `pr#17487 <https://github.com/ceph/ceph/pull/17487>`_, Zac Medico)
+* core: osd crush rule rename not idempotent (`issue#21162 <http://tracker.ceph.com/issues/21162>`_, `pr#17481 <https://github.com/ceph/ceph/pull/17481>`_, xie xingguo)
+* core: osd/PGLog: write only changed dup entries (`issue#21026 <http://tracker.ceph.com/issues/21026>`_, `pr#17378 <https://github.com/ceph/ceph/pull/17378>`_, Josh Durgin)
+* doc: doc/rbd: iSCSI Gateway Documentation (`issue#20437 <http://tracker.ceph.com/issues/20437>`_, `pr#17381 <https://github.com/ceph/ceph/pull/17381>`_, Aron Gunn, Jason Dillaman)
+* luminous: os/bluestore/BlueFS.cc: 1255: FAILED assert(!log_file->fnode.extents.empty()) (`issue#21250 <http://tracker.ceph.com/issues/21250>`_, `pr#17562 <https://github.com/ceph/ceph/pull/17562>`_, Sage Weil)
```
Comment (smithfarm) [inline doc/release-notes.rst]: remove "luminous: " prefix and replace with "bluestore: ", move up to bluestore section

File: doc/release-notes.rst [hunk 5]
```diff
@@ -2,6 +2,123 @@
...
+* luminous: [rbd] image-meta list does not return all entries (`issue#21179 <http://tracker.ceph.com/issues/21179>`_, `pr#17561 <https://github.com/ceph/ceph/pull/17561>`_, Jason Dillaman)
```
Comment (smithfarm) [inline doc/release-notes.rst]: remove "luminous: " prefix, replace with "rbd: " and move to rbd section

File: doc/release-notes.rst [hunk 6]
```diff
@@ -2,6 +2,123 @@
...
+* build/ops: RHEL 7.3 Selinux denials at OSD start (`issue#19200 <http://tracker.ceph.com/issues/19200>`_, `pr#17468 <https://github.com/ceph/ceph/pull/17468>`_, Boris Ranto)
+* build/ops: rocksdb,cmake:  build portable binaries (`issue#20529 <http://tracker.ceph.com/issues/20529>`_, `pr#17745 <https://github.com/ceph/ceph/pull/17745>`_, Kefu Chai)
+* cephfs: client/mds has wrong check to clear S_ISGID on chown (`issue#21004 <http://tracker.ceph.com/issues/21004>`_, `pr#17471 <https://github.com/ceph/ceph/pull/17471>`_, Patrick Donnelly)
+* cephfs: client: missing space in some client debug log messages (`issue#21069 <http://tracker.ceph.com/issues/21069>`_, `pr#17469 <https://github.com/ceph/ceph/pull/17469>`_, Zhu Shangzhong)
+* cephfs: get_quota_root sends lookupname op for every buffered write (`issue#20945 <http://tracker.ceph.com/issues/20945>`_, `pr#17473 <https://github.com/ceph/ceph/pull/17473>`_, Dan van der Ster)
+* cephfs: Incorrect grammar in FS message 1 filesystem is have a failed mds daemon (`issue#21153 <http://tracker.ceph.com/issues/21153>`_, `pr#17566 <https://github.com/ceph/ceph/pull/17566>`_, John Spray)
```
Comment (smithfarm) [inline doc/release-notes.rst]: grammar correction is not a "notable change" - drop

File: doc/release-notes.rst [hunk 7]
```diff
@@ -2,6 +2,123 @@
...
+* bluestore: osd crash when change option bluestore_csum_type from none to CRC32 (`issue#21175 <http://tracker.ceph.com/issues/21175>`_, `pr#17497 <https://github.com/ceph/ceph/pull/17497>`_, xie xingguo)
+* build/ops: ceph-fuse RPM should require fusermount (`issue#21057 <http://tracker.ceph.com/issues/21057>`_, `pr#17470 <https://github.com/ceph/ceph/pull/17470>`_, Ken Dreyer)
+* build/ops: RHEL 7.3 Selinux denials at OSD start (`issue#19200 <http://tracker.ceph.com/issues/19200>`_, `pr#17468 <https://github.com/ceph/ceph/pull/17468>`_, Boris Ranto)
+* build/ops: rocksdb,cmake:  build portable binaries (`issue#20529 <http://tracker.ceph.com/issues/20529>`_, `pr#17745 <https://github.com/ceph/ceph/pull/17745>`_, Kefu Chai)
+* cephfs: client/mds has wrong check to clear S_ISGID on chown (`issue#21004 <http://tracker.ceph.com/issues/21004>`_, `pr#17471 <https://github.com/ceph/ceph/pull/17471>`_, Patrick Donnelly)
+* cephfs: client: missing space in some client debug log messages (`issue#21069 <http://tracker.ceph.com/issues/21069>`_, `pr#17469 <https://github.com/ceph/ceph/pull/17469>`_, Zhu Shangzhong)
```
Comment (smithfarm) [inline doc/release-notes.rst]: whitespace change is not a "Notable change" - drop

File: doc/release-notes.rst [hunk 8]
```diff
@@ -2,6 +2,123 @@
...
+* cephfs: Incorrect grammar in FS message 1 filesystem is have a failed mds daemon (`issue#21153 <http://tracker.ceph.com/issues/21153>`_, `pr#17566 <https://github.com/ceph/ceph/pull/17566>`_, John Spray)
+* cephfs: MDCache::try_subtree_merge() may print N^2 lines of debug message (`issue#21221 <http://tracker.ceph.com/issues/21221>`_, `pr#17712 <https://github.com/ceph/ceph/pull/17712>`_, Patrick Donnelly)
+* cephfs: MDS rank add/remove log messages say wrong number of ranks (`issue#21421 <http://tracker.ceph.com/issues/21421>`_, `pr#17887 <https://github.com/ceph/ceph/pull/17887>`_, John Spray)
+* cephfs: MDS: standby-replay mds should avoid initiating subtree export (`issue#21378 <http://tracker.ceph.com/issues/21378>`_, `issue#21222 <http://tracker.ceph.com/issues/21222>`_, `pr#17714 <https://github.com/ceph/ceph/pull/17714>`_, "Yan, Zheng", Jianyu Li)
+* cephfs: tests: FS_DEGRADED spurious health warnings in some sub-suites (`issue#20892 <http://tracker.ceph.com/issues/20892>`_, `pr#17474 <https://github.com/ceph/ceph/pull/17474>`_, "Yan, Zheng")
```
Comment (smithfarm) [inline doc/release-notes.rst]: test PRs are not "Notable changes" for users - drop

File: doc/release-notes.rst [hunk 9]
```diff
@@ -2,6 +2,123 @@
...
+* cephfs: the standbys are not updated via ceph tell mds.\* command (`issue#21230 <http://tracker.ceph.com/issues/21230>`_, `pr#17565 <https://github.com/ceph/ceph/pull/17565>`_, Kefu Chai)
+* common: adding line break at end of some cli results (`issue#21019 <http://tracker.ceph.com/issues/21019>`_, `pr#17467 <https://github.com/ceph/ceph/pull/17467>`_, songweibin)
```
Comment (smithfarm) [inline doc/release-notes.rst]: whitespace change is not a "Notable change" - drop

File: doc/release-notes.rst [hunk 10]
```diff
@@ -2,6 +2,123 @@
...
+* mgr: ceph mgr versions shows active mgr as Unknown (`issue#21260 <http://tracker.ceph.com/issues/21260>`_, `pr#17635 <https://github.com/ceph/ceph/pull/17635>`_, John Spray)
+* mgr: Crash in MonCommandCompletion (`issue#21157 <http://tracker.ceph.com/issues/21157>`_, `pr#17483 <https://github.com/ceph/ceph/pull/17483>`_, John Spray)
+* mon: mon/OSDMonitor: deleting pool while pgs are being created leads to assert(p != pools.end) in update_creating_pgs() (`issue#21309 <http://tracker.ceph.com/issues/21309>`_, `pr#17634 <https://github.com/ceph/ceph/pull/17634>`_, Joao Eduardo Luis)
+* mon: OSDMonitor: osd pool application get support (`issue#20976 <http://tracker.ceph.com/issues/20976>`_, `pr#17472 <https://github.com/ceph/ceph/pull/17472>`_, xie xingguo)
+* mon: rate limit on health check update logging (`issue#20888 <http://tracker.ceph.com/issues/20888>`_, `pr#17500 <https://github.com/ceph/ceph/pull/17500>`_, John Spray)
+* os/bluestore: fix deferred write deadlock, aio short return handling (`issue#21171 <http://tracker.ceph.com/issues/21171>`_, `pr#17601 <https://github.com/ceph/ceph/pull/17601>`_, Sage Weil)
```
Comment (smithfarm) [inline doc/release-notes.rst]: drop the leading "os/" and move up to bluestore section

File: doc/release-notes.rst [hunk 11]
```diff
@@ -2,6 +2,123 @@
...
+* osd: request new map from PG when needed (`issue#21428 <http://tracker.ceph.com/issues/21428>`_, `pr#17796 <https://github.com/ceph/ceph/pull/17796>`_, Josh Durgin)
+* osd: Revert "osd/OSDMap: allow bidirectional swap of pg-upmap-items" (`issue#21410 <http://tracker.ceph.com/issues/21410>`_, `pr#17812 <https://github.com/ceph/ceph/pull/17812>`_, Sage Weil)
+* osd: subscribe to new osdmap while waiting_for_healthy (`issue#21121 <http://tracker.ceph.com/issues/21121>`_, `pr#17498 <https://github.com/ceph/ceph/pull/17498>`_, Sage Weil)
+* osd: update info only if new_interval (`issue#21203 <http://tracker.ceph.com/issues/21203>`_, `pr#17622 <https://github.com/ceph/ceph/pull/17622>`_, Kefu Chai)
+* pybind: dashboard usage graph getting bigger and bigger (`issue#20746 <http://tracker.ceph.com/issues/20746>`_, `pr#17486 <https://github.com/ceph/ceph/pull/17486>`_, Yixing Yan)
+* qa/objectstore/bluestore\*: less debug output (`issue#20910 <http://tracker.ceph.com/issues/20910>`_, `pr#17564 <https://github.com/ceph/ceph/pull/17564>`_, Sage Weil)
```
Comment (smithfarm) [inline doc/release-notes.rst]: not notable - drop

File: doc/release-notes.rst [hunk 12]
```diff
@@ -2,6 +2,123 @@
...
+* tests: test/cls_rbd: mask newer feature bits to support upgrade tests (`issue#21217 <http://tracker.ceph.com/issues/21217>`_, `pr#17551 <https://github.com/ceph/ceph/pull/17551>`_, Jason Dillaman)
+* tests: test_misc creates metadata pool with dummy object resulting in WRN: POOL_APP_NOT_ENABLED (`issue#21071 <http://tracker.ceph.com/issues/21071>`_, `pr#17879 <https://github.com/ceph/ceph/pull/17879>`_, Douglas Fuller)
+* tests: test_rebuild_simple_altpool triggers MDS assertion (`issue#20337 <http://tracker.ceph.com/issues/20337>`_, `issue#21466 <http://tracker.ceph.com/issues/21466>`_, `pr#17855 <https://github.com/ceph/ceph/pull/17855>`_, Patrick Donnelly, Douglas Fuller)
+* tests: thrashosds read error injection doesn't take live_osds into account (`issue#21206 <http://tracker.ceph.com/issues/21206>`_, `pr#17495 <https://github.com/ceph/ceph/pull/17495>`_, Sage Weil)
+* tests: various teuthology errors (`issue#21251 <http://tracker.ceph.com/issues/21251>`_, `pr#17560 <https://github.com/ceph/ceph/pull/17560>`_, Jason Dillaman)
```
Comment (smithfarm) [inline doc/release-notes.rst]: drop all the items that start with "test: "

**Comment by theanalyst** [comment]
@dillaman is there any announcements on the RBD front, iirc there was some note you wanted to make

**Comment by dillaman** [comment]
@theanalyst Need to merge #17755

File: doc/release-notes.rst [hunk 13]
```diff
@@ -2,6 +2,104 @@
...
+  make the number of PGs per OSD exceed the configured
+  ``mon_max_pg_per_osd`` limit.  The option can be adjusted if it
+  is really necessary to create a pool with more PGs.
+
+* There was a bug in the PG mapping behavior of the new *upmap*
+  feature. If you made use of this feature (e.g., via the `ceph osd
+  pg-upmap-items` command), we recommend that all mappings be removed (via
+  the `ceph osd rm-pg-upmap-items` command) before upgrading to this
+  point release.
+
```
Comment (liewegas) [inline doc/release-notes.rst]: * A stall in BlueStore IO submission that was affecting many users has been resolved.

**Comment by liewegas** [comment]
otherwise lgtm!  update to releases.rst should follow too?

File: doc/release-notes.rst (hunk 13 above)
Comment (theanalyst) [reply doc/release-notes.rst]: ack

**Comment by theanalyst** [comment]
jenkins render docs

**Comment by theanalyst** [comment]
jenkins render docs

File: doc/releases.rst [hunk 14]
```diff
@@ -172,6 +174,7 @@ Timeline
 
 .. _Luminous: ../release-notes#v12-2-0-luminous
 .. _12.2.0: ../release-notes#v12-2-0-luminous
+.. _12.2.1: ../release-notes#v12-2-1-luminous
```
Comment (smithfarm) [inline doc/releases.rst]: nit: move above `_12.2.0` line for consistency with the rest

Comment (smithfarm) [reply doc/releases.rst]: (the entire list is in reverse order)

**Pull request merged**